### PR TITLE
Adds signals for right clicking, right click opens storage items instead of alt click

### PR DESCRIPTION
**Pull request by TJatPBnJ**
## About The Pull Request
Adds signals for right click. Alt click no longer opens storage items like boxes, right click now does.

## Why It's Good For The Game
It reinforces right click as the secondary action key, by adding it to one of the most common actions in the game. 

## Changelog
:cl:
qol: Storage items can be opened with right click, alt click no longer works for this.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
I'm cool with this, but what's the reason for removing alt-click from storage before we have anything that utilizes it? Design space?

**Comment by Angustmeta** [comment]
Can you still splash liquids onto containers/wall safes?

**Comment by Qustinnus** [comment]
also why wouldn't this use secondary_attack_hand which has built in safety checks for distance and stuff

**Comment by TJatPBnJ** [comment]
> also why wouldn't this use secondary_attack_hand which has built in safety checks for distance and stuff

Because I misunderstood how components worked when I first read this. Thanks for letting me know.

**Comment by TJatPBnJ** [comment]
> I'm cool with this, but what's the reason for removing alt-click from storage before we have anything that utilizes it? Design space?

I don't think we need two controls for the same action, and half the reason I made this pr was to reinforce right click as the secondary action key, which, for most things, is still alt-click. This change isn't that hard for old players to adapt to.

**Comment by Mothblocks** [comment]
@Qustinnus Because attack_hand_secondary is only when you have an empty hand, but you can presently alt-click storage components with your hands full, something I would like to keep.

**Comment by TJatPBnJ** [comment]
> Can you still splash liquids onto containers/wall safes?

Thanks for mentioning this. It should be fixed now.

File: code/datums/components/storage/storage.dm [hunk 1]
```diff
@@ -829,10 +828,14 @@
 	return hide_from(target)
 
 /datum/component/storage/proc/on_right_click(datum/source, mob/user)
-	SIGNAL_HANDLER
+	SIGNAL_HANDLER_DOES_SLEEP
```
Comment (Mothblocks) [inline code/datums/components/storage/storage.dm:831]: No new sleeping signal handlers are allowed. Yes, I know you're transferring this over, but no better time than now--move the sleeping part (probably put_in_hands) and everything after in its own proc and invoke that asyncly with INVOKE_ASYNC.

File: code/datums/components/storage/storage.dm [hunk 2]
```diff
@@ -848,7 +848,7 @@
 		return
 	A.add_fingerprint(user)
 	remove_from_storage(I, get_turf(user))
-	if(!user.put_in_hands(I))
+	if(INVOKE_ASYNC(!user.put_in_hands(I)))
```
Comment (Mothblocks) [inline code/datums/components/storage/storage.dm]: Put all the code from this point onwards in its own proc, and INVOKE_ASYNC call that--checking the return of an INVOKE_ASYNC like this doesn't make too much sense in this context.

Comment (TJatPBnJ) [reply code/datums/components/storage/storage.dm]: I've already done this, you're looking at an older version.

Comment (Mothblocks) [reply code/datums/components/storage/storage.dm]: ah

File: code/datums/components/storage/storage.dm [hunk 3]
```diff
@@ -848,10 +848,13 @@
 		return
 	A.add_fingerprint(user)
 	remove_from_storage(I, get_turf(user))
-	if(!user.put_in_hands(I))
-		to_chat(user, "<span class='notice'>You fumble for [I] and it falls on the floor.</span>")
+	INVOKE_ASYNC(src, .proc/quickdraw.quickdraw, user, I)
```
Comment (Mothblocks) [inline code/datums/components/storage/storage.dm:851]: I have no idea what `.proc/quickdraw.quickdraw` is, you should be able to just use `.proc/quickdraw`?

Comment (TJatPBnJ) [reply code/datums/components/storage/storage.dm]: it was a typo

**Comment by MrStonedOne** [comment]
> alt click no longer works for this.

This is not a sane way to move controls around.

Keep alt click but make it print a message to chat that its moving to right click.

Remove alt click in a bit.

**Comment by TJatPBnJ** [comment]
I'm willing to do this, but I'm also certain that anyone playing the game will be able to relearn it on their own. Usually they'll ask/complain in OOC, and get told the new controls by someone who has read the changelog.

**Comment by MrStonedOne** [comment]
Thats not a good user experience thou.

**Pull request closed without merging**